### PR TITLE
FEAT-#6325: Add GPU execution option for HDK backend

### LINKED ISSUE #6325: HDK: Add support for GPU execution flow
PyHDK v0.7.0 introduces an option to run a computation on an Intel GPU. This is to cover Modin support for the GPU mode.

### PR DESCRIPTION
**Pull request by kurapov-peter**
## What do these changes do?

Add an option to run Modin on GPU via HDK. The behavior is controlled by an existing `cpu_only` config option as described in the updated doc.

- [x] first commit message and PR title follow format outlined [here](https://modin.readthedocs.io/en/latest/development/contributing.html#commit-message-formatting)
  > **_NOTE:_**  If you edit the PR title to match this format, you need to add another commit (even if it's empty) or amend your last commit for the CI job that checks the PR title to pick up the new PR title.
- [ ] passes `flake8 modin/ asv_bench/benchmarks scripts/doc_checker.py`
- [ ] passes `black --check modin/ asv_bench/benchmarks scripts/doc_checker.py`
- [ ] signed commit with `git commit -s` 
- [ ] Resolves #6325 
- [ ] tests added and passing
- [ ] module layout described at `docs/development/architecture.rst` is up-to-date

### CODE REVIEW

File: docs/development/using_hdk.rst [hunk 1]
```diff
@@ -34,4 +34,22 @@ If for some reasons ``Native`` engine is explicitly set using ``modin.config`` o
...
+Prerequisites:
+* HDK’s GPU mode is currently supported on Linux only.
+* HDK supports Gen9 architecture and higher (including Xe & Arc).
+* HDK's GPU mode requires proper driver installation. Follow this guide_ to set up your system. Make sure to install the compute runtime packages: ``intel-opencl-icd``, ``intel-level-zero-gpu``, ``level-zero``.
+* Make sure your GPU is visible and accessible.
+
+.. note::
+   You can use ``hwinfo`` and ``clinfo`` utilities to verify the driver installation and device accessibility.
+
+HDK supports a heterogeneous execution mode (experimental) that is disabled by default in Modin. Starting with pyHDK version 0.8 Modin can run the workload on Intel GPU.
```
Comment (Garra1980) [inline docs/development/using_hdk.rst:50]: starting from 0.7 I presume

**Review by Garra1980 — APPROVED** [review]

**Comment by Garra1980** [comment]
LGTM once tests passed and my comment fixed

File: docs/development/using_hdk.rst [hunk 2]
```diff
@@ -34,4 +34,22 @@ If for some reasons ``Native`` engine is explicitly set using ``modin.config`` o
...
+Run on a GPU via ``MODIN_HDK_LAUNCH_PARAMETERS="cpu_only=0" python <your-script.py>``.
```
Comment (YarShev) [inline docs/development/using_hdk.rst:51]:
```suggestion
Run on a GPU via ``MODIN_HDK_LAUNCH_PARAMETERS="cpu_only=0"``.
```
Should this be so? Did you check if it works?

File: docs/development/using_hdk.rst [hunk 3]
```diff
@@ -34,4 +34,22 @@ If for some reasons ``Native`` engine is explicitly set using ``modin.config`` o
...
+
+
```
Comment (YarShev) [inline docs/development/using_hdk.rst:53]:
```suggestion
```

File: modin/experimental/core/execution/native/implementations/hdk_on_native/hdk_worker.py [hunk 4]
```diff
@@ -39,8 +40,11 @@ def setup_engine(cls):
         Do nothing if it is initiliazed already.
         """
         if cls._executor is None:
+            cls._preffered_device = (
+                "CPU" if bool(HdkLaunchParameters.get()["cpu_only"]) else "GPU"
+            )
             cls._config = pyhdk.buildConfig(**HdkLaunchParameters.get())
-            cls._storage = pyhdk.storage.ArrowStorage(1)
+            cls._storage = pyhdk.storage.ArrowStorage(1, cls._config)
```
Comment (YarShev) [inline modin/experimental/core/execution/native/implementations/hdk_on_native/hdk_worker.py:47]: Why did we not pass the config before?

File: docs/development/using_hdk.rst (hunk 1 above)
Comment (kurapov-peter) [reply docs/development/using_hdk.rst]: Thanks, fixed

File: docs/development/using_hdk.rst [hunk 5]
```diff
@@ -34,4 +34,22 @@ If for some reasons ``Native`` engine is explicitly set using ``modin.config`` o
...
 
-.. _HDK: https://github.com/intel-ai/hdk
\ No newline at end of file
+
+Running on a GPU
+---
+
+Prerequisites:
+* HDK’s GPU mode is currently supported on Linux only.
+* HDK supports Gen9 architecture and higher (including Xe & Arc).
```
[the remark this answers is not in the document]
Comment (kurapov-peter) [reply docs/development/using_hdk.rst:43]: Added a note on GPU vendor to prerequisites as well.

File: docs/development/using_hdk.rst (hunk 2 above)
Comment (kurapov-peter) [reply docs/development/using_hdk.rst:51]: Yes. I did not include the suggestion as it is not an executable line and can be confusing.

File: modin/experimental/core/execution/native/implementations/hdk_on_native/hdk_worker.py (hunk 4 above)
Comment (kurapov-peter) [reply modin/experimental/core/execution/native/implementations/hdk_on_native/hdk_worker.py:47]: No idea, our API expects a config. Maybe @AndreyPavlenko knows?

File: docs/development/using_hdk.rst [hunk 6]
```diff
@@ -34,4 +34,22 @@ If for some reasons ``Native`` engine is explicitly set using ``modin.config`` o
    If you encounter ``LLVM ERROR: inconsistency in registered CommandLine options`` error when using HDK,
    please refer to the respective section in :doc:`Troubleshooting </getting_started/troubleshooting>` page to avoid the issue.
 
-.. _HDK: https://github.com/intel-ai/hdk
\ No newline at end of file
+
+Running on a GPU
+---
```
[the remark this answers is not in the document]
Comment (kurapov-peter) [reply docs/development/using_hdk.rst]: Renders correctly now, thanks

File: docs/development/using_hdk.rst [hunk 7]
```diff
@@ -34,4 +34,20 @@ If for some reasons ``Native`` engine is explicitly set using ``modin.config`` o
    If you encounter ``LLVM ERROR: inconsistency in registered CommandLine options`` error when using HDK,
    please refer to the respective section in :doc:`Troubleshooting </getting_started/troubleshooting>` page to avoid the issue.
 
-.. _HDK: https://github.com/intel-ai/hdk
\ No newline at end of file
+
+Running on a GPU
+----------------
+Prerequisites:
+* HDK’s GPU mode is currently supported on Linux and Intel GPU only.
```
Comment (YarShev) [inline docs/development/using_hdk.rst:41]:
```suggestion
----------------

Prerequisites:
* HDK's GPU mode is currently supported on Linux and Intel GPU only.
```

**Review by YarShev — APPROVED** [review]

File: docs/development/using_hdk.rst (hunk 7 above)
Comment (kurapov-peter) [reply docs/development/using_hdk.rst]: Leaving a space in between did not render properly for me if it wasn't a synchronization problem

Comment (YarShev) [reply docs/development/using_hdk.rst]: It is rendered well https://modin--6326.org.readthedocs.build/en/6326/development/using_hdk.html#running-on-a-gpu.

File: docs/development/using_hdk.rst [hunk 8]
```diff
@@ -34,4 +34,21 @@ If for some reasons ``Native`` engine is explicitly set using ``modin.config`` o
...
    please refer to the respective section in :doc:`Troubleshooting </getting_started/troubleshooting>` page to avoid the issue.
 
-.. _HDK: https://github.com/intel-ai/hdk
\ No newline at end of file
+
+Running on a GPU
+----------------
+
+Prerequisites:
+* HDK's GPU mode is currently supported on Linux and Intel GPU only.
```
Comment (YarShev) [inline docs/development/using_hdk.rst:42]:
```suggestion
Prerequisites:

* HDK's GPU mode is currently supported on Linux and Intel GPU only.
```

Comment (YarShev) [reply docs/development/using_hdk.rst:42]: We still need a space here to render a list of bullets properly.

File: docs/development/using_hdk.rst (hunk 7 above)
Comment (kurapov-peter) [reply docs/development/using_hdk.rst]: Yeah, I noticed, guess it was a sync issue then

File: modin/experimental/core/execution/native/implementations/hdk_on_native/hdk_worker.py (hunk 4 above)
Comment (AndreyPavlenko) [reply modin/experimental/core/execution/native/implementations/hdk_on_native/hdk_worker.py:47]: I suppose, it was not required for the previous implementation.
Btw, this code will be removed when #6286 is meged. The new Implementation will use the HDK class which also passes the config to ArrowStorage() - https://github.com/intel-ai/hdk/blob/main/python/pyhdk/hdk.py#L1879

**Review by YarShev — APPROVED** [review]

**Pull request merged**